File: cpp/restaurant.cpp
#include <iostream>
using namespace std;

int gcd(int a, int b) {
    if (a == 0) {
        return b;
    } else if (b == 0) {
        return a;
    } else {
        return gcd(b, a % b);
    }
}

int main() {
    int n;
    cin >> n;

    while (n--) {
        int a, b;
        cin >> a >> b;

        cout << (a * b) / (gcd(a, b) * gcd(a, b)) << "\n";
    }

    return 0;
}

File: java/src/Restaurant.java
// https://www.hackerrank.com/challenges/restaurant
// t: number of test cases
// n: max of l and b in each entry
// T: O(t * log(n))
// S: O(1)

import java.util.Scanner;

public class Restaurant {
    private static final Scanner SCANNER = new Scanner(System.in);

    public static void main(String[] args) {
        int testCases = SCANNER.nextInt();
        while (testCases-- > 0) {
            final int l = SCANNER.nextInt();
            final int b = SCANNER.nextInt();
            System.out.println(maximumNumberOfSquares(l, b));
        }
    }

    private static int maximumNumberOfSquares(int l, int b) {
        final int sideLength = gcd(l, b);
        return (l * b) / (sideLength * sideLength);
    }

    private static int gcd(int a, int b) {
        return b == 0 ? a : gcd(b, a % b);
    }
}
